state/application.go

```diff
@@ -1005,6 +1005,117 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string, force bool) (err error) {
+	if a.Series() == series {
+		return jujutxn.ErrNoOperations
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			// If we've tried once already and failed, re-evaluate the criteria.
+			if err := a.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+			// Exit early if the Application series doesn't need to change.
+			if a.Series() == series {
+				return nil, jujutxn.ErrNoOperations
+			}
+		}
+
+		// Verify and gather data for the transaction operations.
+		err := a.VerifySupportedSeries(series, force)
+		if err != nil {
+			return nil, err
+		}
+		units, err := a.AllUnits()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		var subApps []*Application
+		var unit *Unit
+
+		if len(units) > 0 {
+			// All units have the same subordinates...
+			unit = units[0]
+			for _, n := range unit.SubordinateNames() {
+				app, err := a.st.Application(UnitAppName(n))
+				if err != nil {
+					return nil, err
+				}
+				err = app.VerifySupportedSeries(series, force)
+				if err != nil {
+					return nil, err
+				}
+				subApps = append(subApps, app)
+			}
+		}
+
+		//Create the transaction operations
+		ops, err := a.addApplicationUpdateSeriesOps(series, force)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if unit != nil {
+			ops = append(ops, txn.Op{
+				C:      unitsC,
+				Id:     unit.doc.DocID,
+				Assert: bson.D{{"subordinates", unit.SubordinateNames()}},
+			})
+		}
+		for _, sub := range subApps {
+			appOps, err := sub.addApplicationUpdateSeriesOps(series, force)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			ops = append(ops, appOps...)
+		}
+		return ops, nil
+	}
+
+	err = a.st.run(buildTxn)
+	if err != nil && IsIncompatibleSeriesError(err) {
+		return errors.Annotatef(err, "cannot update series for %q to %s", a, series)
+	}
+	return err
+}
+
+func (a *Application) addApplicationUpdateSeriesOps(series string, force bool) ([]txn.Op, error) {
+	ops := []txn.Op{{
+		C:      applicationsC,
+		Id:     a.doc.DocID,
+		Assert: isAliveDoc,
+		Update: bson.D{{"$set", bson.D{{"series", series}}}},
+	}}
+	ops = append(ops, txn.Op{
+		C:      applicationsC,
+		Id:     a.doc.DocID,
+		Assert: bson.D{{"charmurl", a.doc.CharmURL}},
+	})
+	ops = append(ops, txn.Op{
+		C:      applicationsC,
+		Id:     a.doc.DocID,
+		Assert: bson.D{{"unitcount", a.doc.UnitCount}},
```

Comment:
Same with this asset - add it to the first txn.Op.
So in fact we only need the one txn.Op in the return result. That op will have 3 assert values appended together